### DISCUSSION TITLE
Is it possible to automatically adjust the size of the load increments in strain targeting?

### DISCUSSION THREAD

**Discussion by RunguangLi** [neperfepx/FEPX#85] [Q&A] [answered]
Hi everyone,

I wonder if it is possible to automatically adjust the size of the load increments in strain targeting mode just as abaqus does. For example in my case, what I want is data on the 16% tensile deformed state in tension, but the code often runs into non-convergence conditions unless I try many times and handle every load increment very carefully or small enough. This process is quite time-consuming for million-scale models.

Is there any suggestion or configuration code for the case that can help？
Thanks.

**Comment by mkasemer** [comment, marked as answer]
Adaptive time stepping is not currently implemented in strain-targeting mode in FEPX, and there are currently no plans to do so. I would suggest implementing yourself if it is of prime importance. 

However: convergence issues are generally avoidable: e.g., you are taking too-large time steps, or the mesh is poorly formed. For the former, it is always important to run development simulations on small scale meshes in order to find an appropriate set of simulation parameters, rather than iterating on large scale meshes, which is indeed very time consuming (i.e., run the "big mesh" only when you're certain about the configuration file). This will help you gain intuition about what time steps are too large for your material system. For the latter, the biggest remedy is to use tessellation regularization, which will lead to a much more robust mesh that has fewer errors at large strains.

Philosophically speaking, I'm not a fan of adaptive time stepping, as the user should be explicit about - and the software should obey - the input.
